R/read_redcap.R

```diff
@@ -72,6 +72,14 @@ read_redcap <- function(redcap_uri,
                         forms = NULL,
                         export_survey_fields = TRUE,
                         suppress_redcapr_messages = TRUE) {
+
+  check_arg_is_character(redcap_uri, len = 1)
+  check_arg_is_character(token, len = 1)
```

Comment:
For checking the token for validity, please take a look at https://github.com/OuhscBbmc/REDCapR/blob/main/R/sanitize-token.R. I think we might want to wrap this into its own check, e.g. `check_arg_is_valid_token()`

Reply:
It looks like the only additional thing `sanitize_token()` adds is the check that the token is length 32 hexadecimal but allows the user to override this check with the `REDCAP_TOKEN_PATTERN` environment variable. It seems like we'd need to duplicate that in a `check_arg_is_valid_token()`.

Are we sure we want to do that? It seems like duplication of custom code across packages that we'd want to always be in sync anyways.

Reply:
I would, actually, because I think it should be the responsibility of the exported function to check args for validity. And I'm not talking about reimplementing `sanitize_token()` but actually calling it from our check function.

Reply:
Ah! I didn't realize `sanitize_token()` was exported. That's why I was confused